src/main/java/com/griddynamics/scenario/jagger/JHttpUserScenarioInvoker.java

```diff
@@ -26,29 +25,41 @@ public JHttpUserScenarioInvocationResult invoke(Void nothing, JHttpUserScenario
         Boolean scenarioSucceeded = true;
         JHttpUserScenarioStep previousStep = null;
         List<JHttpUserScenarioStepInvocationResult> stepInvocationResults = new ArrayList<>();
-        JHttpEndpoint globalEndpoint = scenario.getGlobalEndpoint();
 
-        for (JHttpUserScenarioStep userScenarioStep : scenario.userScenarioSteps) {
+        JHttpScenarioGlobalContext localScenarioContext = scenario.getScenarioGlobalContext();
```

Comment:
we need to make a copy here
scenario.getScenarioGlobalContext() should be not changed during scenario execution
we should start scenario with the same values every time